src/PlatformSockets.c

```diff
@@ -448,6 +469,9 @@ SOCKET connectTcpSocket(struct sockaddr_storage* dstaddr, SOCKADDR_LEN addrlen,
         // The socket was signalled
         SOCKADDR_LEN len = sizeof(err);
         getsockopt(s, SOL_SOCKET, SO_ERROR, (char*)&err, &len);
+#ifdef __3DS__ //SO_ERROR is unreliable on 3DS
+        if (err == -26) err = 0;
```

Comment:
Will the 3DS just do a blocking `connect()` (so the `poll()` afterwards isn't really doing anything), or does it actually properly perform a non-blocking `connect()` but then just have no way to report the success/failure?

If it's the former, let's just `#ifdef` out all this `poll()` and `SO_ERROR` stuff for 3DS. If it's the latter and non-blocking socket support otherwise works okay, maybe we can do a 1 byte `recv()` with `MSG_PEEK` to see if the socket is alive (should return `EAGAIN` or `EWOULDBLOCK`).

Reply:
Looks like non-blocking connect works, so I added the recv test!